src/main/java/io/cdap/plugin/servicenow/ServiceNowBaseConfig.java

```diff
@@ -0,0 +1,224 @@
+  private String clientId;
+
+  @Name(ServiceNowConstants.PROPERTY_CLIENT_SECRET)
+  @Macro
+  @Description("The Client Secret for ServiceNow Instance.")
+  private String clientSecret;
+
+  @Name(ServiceNowConstants.PROPERTY_API_ENDPOINT)
+  @Macro
+  @Description("The REST API Endpoint for ServiceNow Instance. For example, https://instance.service-now.com")
+  private String restApiEndpoint;
+
+  @Name(ServiceNowConstants.PROPERTY_USER)
+  @Macro
+  @Description("The user name for ServiceNow Instance.")
+  private String user;
+
+  @Name(ServiceNowConstants.PROPERTY_PASSWORD)
+  @Macro
+  @Description("The password for ServiceNow Instance.")
+  private String password;
+
+  public ServiceNowBaseConfig(String referenceName, String clientId, String clientSecret, String restApiEndpoint,
+                              String user, String password) {
+
+    this.referenceName = referenceName;
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+    this.restApiEndpoint = restApiEndpoint;
+    this.user = user;
+    this.password = password;
+  }
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
+  public String getRestApiEndpoint() {
+    return restApiEndpoint;
+  }
+
+  public String getUser() {
+    return user;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  /**
+   * Validates {@link ServiceNowSourceConfig} instance.
+   */
+  public void validate(FailureCollector collector) {
+    // Validates the given referenceName to consists of characters allowed to represent a dataset.
+    IdUtils.validateReferenceName(referenceName, collector);
+    validateCredentials(collector);
+  }
+
+  public void validateCredentials(FailureCollector collector) {
+    if (!shouldConnect()) {
+      return;
+    }
+
+    if (Util.isNullOrEmpty(clientId)) {
+      collector.addFailure("Client ID must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_CLIENT_ID);
+    }
+
+    if (Util.isNullOrEmpty(clientSecret)) {
+      collector.addFailure("Client Secret must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_CLIENT_SECRET);
+    }
+
+    if (Util.isNullOrEmpty(restApiEndpoint)) {
+      collector.addFailure("API Endpoint must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_API_ENDPOINT);
+    }
+
+    if (Util.isNullOrEmpty(user)) {
+      collector.addFailure("User name must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_USER);
+    }
+
+    if (Util.isNullOrEmpty(password)) {
+      collector.addFailure("Password must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_PASSWORD);
+    }
+
+    validateServiceNowConnection(collector);
+  }
+  
+  public void validateServiceNowConnection(FailureCollector collector) {
```

Comment:
are we using it in other classes other than test?  If not make it package private